File: backend/n8n/tagger.py
import sys
import mutagen
from mutagen.easyid3 import EasyID3
import os

if len(sys.argv) < 4:
    print("Usage: python3 tagger.py <file_path> <artist> <title>")
    sys.exit(1)

filepath = sys.argv[1]
artist = sys.argv[2]
title = sys.argv[3]

if not os.path.exists(filepath):
    print(f"Error: File not found - {filepath}")
    sys.exit(1)

try:
    audio = EasyID3(filepath)
except mutagen.id3.ID3NoHeaderError:
    # If no ID3 header exists, create one
    audio = mutagen.File(filepath, easy=True)
    audio.add_tags()
except Exception as e:
    # Fallback to standard mutagen
    audio = mutagen.File(filepath, easy=True)
    if audio is None:
        print(f"Error: {filepath} is not a valid or supported audio file.")
        sys.exit(1)
    try:
        audio.add_tags()
    except Exception:
        pass

audio['artist'] = artist
audio['title'] = title
audio.save()

print(f"Success: Tagged '{filepath}' with Artist='{artist}' and Title='{title}'")

File: backend/n8n/tagger.js
const NodeID3 = require('node-id3');
const fs = require('fs');

const args = process.argv.slice(2);
if (args.length < 3) {
    console.error("Usage: node tagger.js <file_path> <artist> <title>");
    process.exit(1);
}

const filepath = args[0];
const artist = args[1];
const title = args[2];

if (!fs.existsSync(filepath)) {
    console.error(`Error: File not found - ${filepath}`);
    process.exit(1);
}

const tags = {
    artist: artist,
    title: title
};

const success = NodeID3.update(tags, filepath);

if (success === true) {
    console.log(`Success: Tagged '${filepath}' with Artist='${artist}' and Title='{title}'`);
} else {
    console.error(`Error: Failed to write tags to '${filepath}'`);
    process.exit(1);
}
